effekt/jvm/src/test/scala/effekt/StdlibTests.scala

```diff
@@ -0,0 +1,48 @@
+package effekt
+
+import java.io.File
+import sbt.io.*
+import sbt.io.syntax.*
+
+import scala.language.implicitConversions
+
+abstract class StdlibTests extends EffektTests {
+  override lazy val included: List[File] = List(
+    examplesDir / "stdlib",
+  )
+
+  override lazy val ignored: List[File] = List()
+}
+
+class StdlibJavaScriptTests extends StdlibTests {
+  def backendName: String = "js"
+}
+class StdlibChezSchemeMonadicTests extends StdlibTests {
+  def backendName: String = "chez-monadic"
+}
+class StdlibChezSchemeCallCCTests extends StdlibTests {
+  def backendName: String = "chez-callcc"
+}
+class StdlibChezSchemeLiftTests extends StdlibTests {
+  def backendName: String = "chez-lift"
+}
+class StdlibMLTests extends StdlibTests {
+  def backendName: String = "ml"
+
+  override lazy val ignored: List[File] = List(
+    // For every function using `foreach`:
+    // [error] Effect polymorphic recursion is not allowed. The following definition is effect polymorphic since unification variable X occurs in instantiation ...
+    examplesDir / "stdlib" / "list" / "join.effekt",
+    examplesDir / "stdlib" / "list" / "flatmap.effekt",
```

Comment:
cc @b-studios 
This is because we use
```
var reminder = l
loop { // recursive from `Control`
  // ...
}
```
but we can't do that in the MLTon backend.

What's the best way to go about this? Should we:
1. not use `Control` at all [or at least not in the MLTon backend]
2. write a better version
3. ignore the issue for now

Reply:
Here's the relevant code I'm talking about:
https://github.com/effekt-lang/effekt/pull/355/files#diff-f8dcc2131b31d921d0b18b09a86024e013b5b6902d7a0dc065a3f71eacf941deR111-R122

Reply:
To be honest, long term I would like to reimplement MLTon's optimizations in our backend and then either move to OCaml or replace it by LLVM / RiscV / Wasm. MLton was a great way of showing what is possible with WPO, but was only ever a means not the end.

Reply:
With #383, we could now make the `Control` code a little simpler, potentially.